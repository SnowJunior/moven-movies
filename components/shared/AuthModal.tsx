import { AuthModalProps } from "@/model/components";
import { signOut } from "next-auth/react";
import React from "react";
import { IoClose } from "react-icons/io5";

const AuthModal: React.FC<AuthModalProps> = ({ onClose }) => {
  return (
    <div className="fixed bg-black inset-0 bg-opacity-50 flex items-center justify-center z-50">
      <div className="bg-white rounded-lg w-[40%] p-12 lg:p-6 relative">
        <button
          onClick={onClose}
          className="absolute top-4 right-4 text-xl font-bold text-gray-700"
        >
          <IoClose size={30} />
        </button>
        <div className="flex flex-col text-black items-center justify-center w-full gap-6">
          <div className="flex flex-col items-center justify-center">
            <p className="text-2xl font-bold">Logout User</p>
            <p className="text-lg">Are you sure you to exit ?</p>
          </div>
          <div className="flex gap-4">
            <button
              onClick={onClose}
              className={`bg-black w-44 flex items-center justify-center gap-3 h-12 rounded-md text-white text-xl`}
            >
              Cancel
            </button>
            <button
              onClick={() => signOut()}
              className={`bg-red-500 w-44 flex items-center justify-center gap-3 h-12 rounded-md text-white text-xl`}
            >
              Logout
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};

export default AuthModal;
