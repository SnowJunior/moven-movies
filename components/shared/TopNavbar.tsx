"use client";
import Image from "next/image";
import React from "react";
import logo from "@/public/images/lightLogo.png";
import { Avatar } from "@heroui/avatar";

const TopNavbar = () => {
  return (
    <div className="h-24 bg-black text-white">
      <div className="flex flex-row w-[90%] mx-auto items-center justify-between mt-4">
        <div className="flex flex-row gap-4 items-center">
          <Image
            className="rounded-[50%]"
            src={logo}
            alt="moven image"
            width={70}
            height={70}
          />
          <h2 className="text-xl font-bold hidden sm:block">Dashboard</h2>
        </div>
        <div className="flex gap-2">
          <div
            className="flex gap-4 items-center"
          >
            <Avatar
              isBordered
              color="success"
              size="lg"
              src={"https://i.pravatar.cc/150?u=a04258114e29026302d"}
              alt="moven image"
            />
            <div className="text-lg">@snow</div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default TopNavbar;
