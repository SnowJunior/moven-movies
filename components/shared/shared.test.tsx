import "@testing-library/jest-dom";
import { render, screen, fireEvent } from "@testing-library/react";

import TopNavbar from "./TopNavbar";
import MovieModal from "./CardModal";
import AuthModal from "./AuthModal";

const mockOnClose = jest.fn();

describe("TopNavbar Component", () => {
  it("renders the top navbar component correctly", () => {
    render(<TopNavbar />);

    expect(screen.getByText("Dashboard")).toBeInTheDocument();
    expect(screen.getByText("@")).toBeInTheDocument();
  });

  it("shows the avatar image", () => {
    render(<TopNavbar />);

    const avatar = screen.findAllByAltText("moven image");
    expect(avatar).toMatchSnapshot();
  });
});

describe("MovieModal", () => {
  const mockMovie = {
    title: "Movie modal test",
    adult: false,
    overview:
      "Kraven Kravinoff's complex relationship with his ruthless gangster father, Nikolai, starts him down a path of vengeance with brutal consequences, motivating him to become not only the greatest hunter in the world, but also one of its most feared.",
    vote_average: 8.5,
    release_date: "2023-10-27",
    popularity: 95.5,
    poster_path: "/i47IUSsN126K11JUzqQIOi1Mg1M.jpg",
    id: 98789,
    backdrop_path: "/v9Du2HC3hlknAvGlWhquRbeifwW.jpg",
  };

  let mockOnClose: jest.Mock;

  beforeEach(() => {
    mockOnClose = jest.fn();
  });

  it("renders the component with movie data and handles image loading", async () => {
    render(
      <MovieModal movie={mockMovie} onClose={mockOnClose} isLoading={false} />
    );

    // Check if movie details are displayed
    expect(
      screen.getByRole("heading", { name: mockMovie.title })
    ).toBeInTheDocument();
    expect(screen.getByText(`Rated:`));
    expect(screen.getByText(`Description:`)).toBeInTheDocument();
    expect(screen.getByText(mockMovie.overview)).toBeInTheDocument();
    expect(screen.getByText(`Viewer rating:`)).toBeInTheDocument();
    expect(screen.getByText(`${mockMovie.vote_average}`)).toBeInTheDocument();
    expect(screen.getByText(`Release date:`)).toBeInTheDocument();
    expect(screen.getByText(mockMovie.release_date)).toBeInTheDocument();
    expect(screen.getByText(`Popularity:`)).toBeInTheDocument();
    expect(screen.getByText(`${mockMovie.popularity}`)).toBeInTheDocument();
  });

  it("calls onClose when close button is clicked", () => {
    render(
      <MovieModal movie={mockMovie} onClose={mockOnClose} isLoading={false} />
    );
    const closeButton = screen.getByRole("button"); // Or query by icon if you prefer
    fireEvent.click(closeButton);
    expect(mockOnClose).toHaveBeenCalledTimes(2);
  });

  test("renders movie poster image", () => {
    render(
      <MovieModal movie={mockMovie} onClose={mockOnClose} isLoading={false} />
    );
    const posterImage = screen.getByTestId("movie-poster");

    expect(posterImage).toBeInTheDocument();
    expect(posterImage.getAttribute("src")).toContain(
      encodeURIComponent(mockMovie.poster_path)
    );
  });

  test("renders fallback image when poster_path is null", () => {
    const movieWithoutPoster = { ...mockMovie, poster_path: null };

    render(
      <MovieModal
        movie={movieWithoutPoster}
        onClose={mockOnClose}
        isLoading={false}
      />
    );

    const posterImage = screen.getByTestId("movie-poster");

    expect(posterImage).toBeInTheDocument();
    expect(posterImage.getAttribute("src")).toContain(
      encodeURIComponent("https://heroui.com/images/hero-card-complete.jpeg")
    );
  });

  test("renders movie backdrop image", () => {
    render(
      <MovieModal movie={mockMovie} onClose={mockOnClose} isLoading={false} />
    );
    const posterImage = screen.getByTestId("mobile-poster");

    expect(posterImage).toBeInTheDocument();
    expect(posterImage.getAttribute("src")).toContain(
      encodeURIComponent(mockMovie.backdrop_path)
    );
  });

  test("renders fallback image when poster_path is null", () => {
    const movieWithoutPoster = { ...mockMovie, backdrop_path: null };

    render(
      <MovieModal
        movie={movieWithoutPoster}
        onClose={mockOnClose}
        isLoading={false}
      />
    );

    const posterImage = screen.getByTestId("mobile-poster");

    expect(posterImage).toBeInTheDocument();
    expect(posterImage.getAttribute("src")).toContain(
      encodeURIComponent("https://heroui.com/images/hero-card-complete.jpeg")
    );
  });
});

describe("AuthModal", () => {
  test("renders Logout text", () => {
    render(<AuthModal onClose={mockOnClose} logOut={mockOnClose} />);
    expect(screen.getByText(/Logout User/i)).toBeInTheDocument();
    expect(screen.getByText(/Are you sure you to exit/i)).toBeInTheDocument();
  });

  test("contains at least one button", () => {
    render(<AuthModal onClose={mockOnClose} />);
    const buttons = screen.getAllByRole("button");
    expect(buttons.length).toBeGreaterThan(0);
  });
});
