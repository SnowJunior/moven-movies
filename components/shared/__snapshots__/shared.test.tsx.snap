// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopNavbar Component shows the avatar image 1`] = `Promise {}`;
